src/js/controller/controller.js

```diff
@@ -978,9 +978,19 @@ Object.assign(Controller.prototype, {
         };
 
         this.setCues = function (cues) {
-            _model.set('cues', cues);
+            let newCues = [];
+            if (cues && cues.length) {
+                // If setCues() is called with empty array, we want to clear the cues.
+                // Otherwise, add the new cues to the existing cues.
+                // This allows setting of cues by ad plugins or setCues() API.
+                const existingCues = _model.get('cues');
+                newCues = existingCues ? existingCues.concat(cues) : cues;
+            }
```

Comment:
This goes back to what the original feedback @pajong and I left - `setCues` should not append to cues.

You could add a new method called `addCues` that does this, and then avoid the repetition of calling `setCues(getCues().append(newCues))` in every place that currently calls `setCues()` to add chapter or ads markers.